contentcuration/contentcuration/serializers.py

```diff
@@ -91,43 +91,35 @@ class FileListSerializer(serializers.ListSerializer):
 
     def update(self, instance, validated_data):
         ret = []
-        update_files = {}
-        user = self.context['request'].user
+        nodes_to_check = []
```

Comment:
Do we POST/PATCH directly to this endpoint, or is the FileListSerializer mostly used within the ContentNode endpoint?

What I did in my updates to the AssessmentItemListSerializer, was to abstract away the need for the frontend to know anything about the implementation of the file objects, and leave that as a book keeping detail on the backend: https://github.com/learningequality/studio/blob/develop/contentcuration/contentcuration/serializers.py#L285

If the same is true here, this could be implemented to require fewer deletes and writes, just pass back the checksums, and language details for the file, and then just update what is associated with the content node.

Reply:
It's for the PUT action only, although I think this might have multiple nodes that can be passed in (e.g. `[node 1- video, node 1- subtitle, node 2- document]`)

That makes sense, going by contentnode rather than files then going back to the nodes again. I'll add it to the tech debt section